presto-thrift-connector/src/main/java/com/facebook/presto/connector/thrift/ThriftHeaderProvider.java

```diff
@@ -19,5 +19,11 @@
 
 public interface ThriftHeaderProvider
 {
-    Map<String, String> getHeaders(ConnectorSession session);
+    @Deprecated
+    default Map<String, String> getHeaders(ConnectorSession session)
+    {
+        return getHeaders(session, null);
+    }
+
+    Map<String, String> getHeaders(ConnectorSession session, ThriftConnectorConfig config);
```

Comment:
I don't think this is a good idea, because `ConnectorSession` often overrides the config, while config serves as the default value. Instead of adding config, should we consider adding those fields into thrift connector session instead?

Reply:
> What is the motivation of the PR? 

The high-level goal is to introduce a change which passes Presto User details to Thrift via a Thrift header ([Commit Here](https://github.com/DerekTBrown/presto/commit/cf01e667e8f5c199939dcb4fb28cbcbe6b602253)).  This is needed for us to apply admission control to users of the Thrift Connected service.

However, we probably don't want to enable this for all existing Thrift Connector services. Hence we needed a connector-level setting to do this.

> If we want to expose some fields, can we use the session property?

My understanding was that `ConnectorSession` is constructed per-user/per-query. Therefore, adding a config option to the session is likely to be more expensive than adding a config option at the connector level.  Let me know if this is the case.

Reply:
I wouldn't worry about the cost of adding a session property. If session property works for your use case. Let's add it. This is standard practice and has been working well.